integration_tests/upsert-avro/datagen.py

```diff
@@ -0,0 +1,83 @@
+from confluent_kafka import Producer
+from confluent_kafka.admin import AdminClient, NewTopic
+from confluent_kafka.serialization import StringSerializer, SerializationContext, MessageField
+from confluent_kafka.schema_registry import SchemaRegistryClient
+from confluent_kafka.schema_registry.avro import AvroSerializer
+import sys
+import random
+import time
+
+
+key_schema = r'''
+"string"
+'''
+value_schema = r'''
+{"type":"record","name":"OBJ_ATTRIBUTE_VALUE","namespace":"CPLM","fields":[{"name":"op_type","type":["null","string"],"default":null},{"name":"ID","type":["null","string"],"default":null},{"name":"CLASS_ID","type":["null","string"],"default":null},{"name":"ITEM_ID","type":["null","string"], "default":null},{"name":"ATTR_ID","type":["null","string"],"default":null},{"name":"ATTR_VALUE","type":["null","string"],"default":null},{"name":"ORG_ID","type":["null","string"],"default":null},{"name":"UNIT_INFO","type":["null","string"],"default":null},{"name":"UPD_TIME","type":["null","string"],"default":null}],"connect.name":"CPLM.OBJ_ATTRIBUTE_VALUE"}
```

Comment:
IIUC, this script only tests the case when the kafka message key is included as part of the message value. There are two missing cases to be tested:
1. Key is not part of the value. We can remove the ID field in the schema to test it.
2. Tombstone value. Thie can be tested by producing a null value to kafka.

Reply:
Both cases are tested in E2E  `scripts/source/test_data/upsert_student_avro_json.1` & `scripts/source/test_data/upsert_student_key_not_subset_of_value_avro_json.1`. 

Since the assertion part of our `integration_tests` is not strong enough, the `integration_test/upsert-avro` is used as an example.

Reply:
> Both cases are tested in E2E `scripts/source/test_data/upsert_student_avro_json.1` & `scripts/source/test_data/upsert_student_key_not_subset_of_value_avro_json.1`.
> 
> Since the assertion part of our `integration_tests` is not strong enough, the `integration_test/upsert-avro` is used as an example.

I see. Thanks for the explanation.